src/webui/prefjson.cpp

```diff
@@ -322,8 +322,12 @@ void prefjson::setPreferences(const QString& json)
         session->setGlobalDownloadSpeedLimit(m["dl_limit"].toInt());
     if (m.contains("up_limit"))
         session->setGlobalUploadSpeedLimit(m["up_limit"].toInt());
-    if (m.contains("enable_utp"))
-        session->setUTPEnabled(m["enable_utp"].toBool());
+    if (m.contains("enable_utp")) {
+        const int value = m["enable_utp"].toInt();
+        if ((static_cast<int>(BitTorrent::BTProtocol::Both) <= value)
```

Comment:
setBTProtocol should check its params instead.

Reply:
This is doing sanitizing it should stay here, no?